### PR TITLE
[Feature/extensions] Adding support for registering Transport Actions for extensions

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Adds support for registering Transport Actions for extensions. 

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-sdk-java/issues/105

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java:74]: Are we having separate action for extensions?

File: server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java
Comment (dbwiddis) [inline server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java:25]: Need a class to test this.  (Should be able to use object equality after writing and reading to a stream.)

File: server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java
Comment (dbwiddis) [inline server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java:26]: Do you want a getter for this?

File: server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java
Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java:74]: There is an open issue: https://github.com/opensearch-project/OpenSearch/issues/4334. Lets do a sweep all together in a dedicated PR.

File: server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java
Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java:26]: Very likely :) 
Its probably coming with: https://github.com/opensearch-project/opensearch-sdk-java/issues/106.

File: server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java
Comment (joshpalis) [inline server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java:42]: Small change, IllegalArguementException should say "Could **not** read transport action"

File: server/src/test/java/org/opensearch/extensions/RegisterTransportActionsRequestTests.java
Comment (dbwiddis) [inline server/src/test/java/org/opensearch/extensions/RegisterTransportActionsRequestTests.java:34]: (Comment) I realize that `close()` is a noop here so these don't need to be closed, but some IDE linters will complain that a resource isn't closed.  Personal preference of mine. :)

**Review by dbwiddis — APPROVED** [review]
LGTM. One optional comment inline

**Comment by saratvemulapalli** [comment]
@owaiskazi19 Waiting for the 2nd review. 
I'll follow up with https://github.com/opensearch-project/opensearch-sdk-java/issues/106 after this set of PRs are merged.

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**
